File: week-04/day-03/Testing/src/Count/CountLettersTest.java
package Count;

import org.junit.Test;

import java.util.HashMap;

import static org.junit.Assert.assertEquals;

public class CountLettersTest {
  HashMap<Character, Integer> characterOccurence;

  @Test
  public void TestWithPanama() {
    characterOccurence = CountLetters.Count("Panama");

    assertEquals(3, (int)characterOccurence.get('a'));
  }

  @Test
  public void TestWithMississippi() {
    characterOccurence = CountLetters.Count("Mississippi");

    assertEquals(4, (int)characterOccurence.get('i'));
    assertEquals(4, (int)characterOccurence.get('s'));
  }

  @Test
  public void TestWithAlabama() {
    characterOccurence = CountLetters.Count("Alabama");

    assertEquals(4, (int)characterOccurence.get('a'));
  }
}

File: CSharp/EntityFramework/EntityTest/CountLettersTest.cs
using System.Collections.Generic;
using EntityFramework;
using NUnit.Framework;

namespace EntityTest
{
    [TestFixture]
    public class CountLettersTest
    {
        private Program testProgram;
        Dictionary<char, int> characterOccurence;

        [SetUp]
        public void Init()
        {
            testProgram = new Program();
        }
        
        [Test]
        public void TestWithPanama()
        {
            characterOccurence = testProgram.Count("Panama");

            Assert.AreEqual(3, characterOccurence['a']);
        }
        
        [Test]
        public void TestWithMississippi()
        {
            characterOccurence = testProgram.Count("Mississippi");

            Assert.AreEqual(4, characterOccurence['i']);
            Assert.AreEqual(4, characterOccurence['s']);
        }
        
        [Test]
        public void TestWithAlabama()
        {
            characterOccurence = testProgram.Count("Alabama");

            Assert.AreEqual(4, characterOccurence['a']);
        }
    }
}